plugins/Synchro/SynchroSynth.h

```diff
@@ -73,6 +73,7 @@ class SynchroSynth : public Instrument
 	virtual QString nodeName() const;
 	virtual f_cnt_t desiredReleaseFrames() const { return(64); } //Not sure what this is used for
 	virtual PluginView * instantiateView(QWidget * parent);
+	virtual Flags flags() const {	return IsSingleStreamed; } //Disables default envelopes/LFOs
```

Comment:
Added this since I plan to add per-oscillator envelopes (exponential, too). However, since I disabled this, LMMS immediately crashes upon playing a note through Synchro. I would guess this is because there's no envelope to process the audio through.

Reply:
It's because `working_buffer` is null in `playNote()` if you specify `IsSingleStreamed`. It's intended to make `playNote()` only check for the status for a note and render the sound in `play()`. See LB302 for how single-streamed instruments work.

Reply:
My intention is to disable the default envelopes as I intend to add my own to the instrument itself. Is there a proper way to do this then?

Reply:
No, at least currently. Do you want to disable LFOs and the filter as well?

Reply:
Yes, though there are a few edge cases where they could be useful. I'll re-enable it for now until I get the custom envelopes worked out.